File: 19.删除链表的倒数第-n-个结点.go
/*
 * @lc app=leetcode.cn id=19 lang=golang
 *
 * [19] 删除链表的倒数第 N 个结点
 */
package main

// @lc code=start
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func removeNthFromEnd(head *ListNode, n int) *ListNode {
	listNodeLen := 1
	for currentListNode := head; currentListNode.Next != nil; currentListNode = currentListNode.Next {
		listNodeLen++
	}
	if listNodeLen < n {
		return head
	}
	if listNodeLen == n {
		return head.Next
	}
	currentListNode := head
	for index := 0; index < listNodeLen-n-1; index++ {
		currentListNode = currentListNode.Next
	}
	tempListNode := currentListNode.Next
	currentListNode.Next = tempListNode.Next
	return head
}

// @lc code=end

File: 19.删除链表的倒数第-n-个结点.cs
using System.Collections.Generic;
/*
 * @lc app=leetcode.cn id=19 lang=csharp
 *
 * [19] 删除链表的倒数第 N 个结点
 */

// @lc code=start
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution
{
    public ListNode RemoveNthFromEnd(ListNode head, int n)
    {
        int len = 1;
        ListNode currentListNode = head;
        for (; currentListNode.next != null; currentListNode = currentListNode.next)
        {
            len++;
        }
        if (len < n) { return head; }
        if (len == n) { return head.next; }
        currentListNode = head;
        for (int i = 0; i < len - n - 1; i++)
        {
            currentListNode = currentListNode.next;
        }
        ListNode tempListNode = currentListNode.next;
        currentListNode.next = tempListNode.next;
        return head;
    }
}
// @lc code=end

